invokeai/backend/model_manager/load/model_cache/model_cache_default.py

```diff
@@ -162,7 +162,8 @@ def put(
         if key in self._cached_models:
             return
         self.make_room(size)
-        cache_record = CacheRecord(key, model, size)
+        state_dict = model.state_dict() if hasattr(model, "state_dict") else None
```

Comment:
The expectation is that this state_dict is on the `self.storage_device` here. To avoid surprising bugs, we should assert that this is True rather than trusting the caller.

Reply:
Done.